### DISCUSSION TITLE
Welcome to pocl Discussions!

### DISCUSSION THREAD

**Discussion by pjaaskel** [pocl/pocl#1145] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MananKansara09** [comment]
heyy how should you implement this pocl how beginner can start to read repo.

**Reply by pjaaskel** [reply]
http://portablecl.org/docs/html/driver_design.html this new dox by @franz should get you started!

**Comment by heweiwill** [comment]
Hi,
Is there any information about how to enhance the performance POCL on Nvidia gpu hardware ?

**Reply by pjaaskel** [reply]
Perhaps @isuruf can give a list of known perf. issues and how to tackle them in the code.

**Reply by isuruf** [reply]
@heweiwill, do you have a specific kernel in mind where the performance is low, or are you looking at improving the performance in general?
If latter, I can think of a few specific things. For eg: pocl uses `sin` to implement `native_sin`, but using `__nv_fast_sinf` is better for performance.

**Reply by heweiwill** [reply]
Hi Isuruf,

Thanks very much for your reply!   I am looking at PoCL running at arm cpu + nvidia gpu, currently we have to apply "-g -cl-opt-disable" to avoid kernel crash.

**Reply by isuruf** [reply]
Please open an issue with more details about the setup.
